Possible dangling reference in ModifySafeMap::get

### Minetest version

```true
Minetest 5.9.0-dev-d2a089ffd-dirty (Windows)
Using LuaJIT 2.1.0-beta3
Built by MSVC 19.39
Running on Windows/10.0.22621 x86_64
BUILD_TYPE=Release
RUN_IN_PLACE=1
USE_CURL=1
USE_GETTEXT=0
USE_SOUND=1
STATIC_SHAREDIR="."
```


### Irrlicht device

_No response_

### Operating system and version

Microsoft Windows 11 Home 10.0.22631 Build 22631

### CPU model

_No response_

### GPU model

_No response_

### Active renderer

_No response_

### Summary

The following snippet is from util/container.h L365-372:
```cpp
        const V &get(const K &key) const {
                if (m_iterating) {
                        auto it = m_new.find(key);
                        if (it != m_new.end())
                                return it->second;
                }
                auto it = m_values.find(key);
                return it == m_values.end() ? null_value : it->second;
        }
```
A ternary operator may create a temporary object if it has to convert one of the operands to a prvalue [according to these rules](https://en.cppreference.com/w/cpp/language/operator_other). If that happens, this function may return a dangling reference to that temporary object. MSVC thinks that's exactly what happens in this method.

> C:\Users\yoshe\code\open_contrib\minetest\src\util\container.h(372): warning C4172: returning address of local variable
 or temporary [C:\Users\yoshe\code\open_contrib\minetest\build-msvc\src\minetest.vcxproj]
 
 I was unable to reproduce the warning on GCC nor do GCC Asan or Valgrind report any errors running the following unit test:
```cpp
void testTemporary() {
        ModifySafeMap<int, int> map;
        map.put(0, 0x12);
        UASSERT(map.get(0) == 0x12);
}
```
I don't have the C++ knowledge to know whether this is a false positive or not. In any case it seems quite risky compared to a simple if/else that definitely won't dangle anything.

### Steps to reproduce

Compile on MSVC and observe the warning.

--- a/src/util/container.h
+++ b/src/util/container.h
@@ -369,7 +369,14 @@ public:
 				return it->second;
 		}
 		auto it = m_values.find(key);
-		return it == m_values.end() ? null_value : it->second;
+		// This conditional block was converted from a ternary to ensure no
+		// temporary values are created in evaluating the return expression,
+		// which could cause a dangling reference.
+		if (it != m_values.end()) {
+			return it->second;
+		} else {
+			return null_value;
+		}
 	}
 
 	void put(const K &key, const V &value) {

